frontend/catalyst/compiler.py

```diff
@@ -147,9 +152,10 @@ class MHLOPass(PassPipeline):
 
     @staticmethod
     def get_output_filename(infile):
-        if not infile.endswith(".mlir"):
-            raise ValueError(f"Input file ({infile}) for MHLO is not an MLIR file")
-        return infile.replace(".mlir", ".nohlo.mlir")
+        path = pathlib.Path(infile)
+        if not path.exists():
+            raise FileNotFoundError("Cannot find {infile}.")
+        return str(path.with_suffix(".nohlo.mlir"))
```

Comment:
The extension holds no special meaning and it can be anything if there's a custom compilation pipeline. So, correct, we don't check for specific extensions ahead of time.

Reply:
The `.mlir` extension? I mean I know extensions aren't enforced in any way, but it does hold meaning

Reply:
They're like variable names. They should be human readable and definitely hold meaning, but the machine doesn't care what the name of the variable is. I like to be liberal in what is accepted by the function. If it can be accepted, it should be accepted. The only real restriction is that the file contains invalid MLIR.